src/test/java/org/cactoos/io/LoggingInputTest.java

```diff
@@ -0,0 +1,218 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link LoggingInput}.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.29
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+@SuppressWarnings(
+    {
+        "PMD.MoreThanOneLogger",
+        "PMD.AvoidDuplicateLiterals"
+    }
+)
+public final class LoggingInputTest {
+
+    @Test
+    public void logReadFromDeadInput() throws IOException {
+        final Logger logger = new FakeLogger();
+        new LengthOf(
+            new LoggingInput(
+                new DeadInput(),
+                "dead input",
+                logger
+            )
+        ).intValue();
```

Comment:
@fabriciofx I see in most tests, you wrap ``LoggingInput`` with ``LengthOf`` and call ``intValue()`` -- I guess intValue() returns the number of bytes read, right? It would be nicer to do an assertion agains this value, instead of (or together with) asserting the text. 

So, for instance, this test would become: 

```java
final int read = new LengthOf(...).intValue();
MatcherAssert.assertThat(read, Matchers.is(0));
MatcherAssert.assertThat(
    "Can't log zero byte read from dead input",
     logger.toString(),
     Matchers.containsString("Read 0 byte(s) from dead input in")
);
```
What do you think?

Reply:
@amihaiemil I think it redundant, because we will check in log the same value. Besides, it's important check multiple messages too.